File: DotNet/Tests/Mocks/MockProvider.cs
using Bollywell.Hydra.Messaging.Storage;
using System;

namespace Bollywell.Hydra.Tests.Mocks
{
    class MockProvider : IProvider
    {
        private readonly IStore _store;

        public MockProvider(IStore store, int? pollIntervalMs = null)
        {
            _store = store;
            PollIntervalMs = pollIntervalMs;
            HydraServer = _store.Name;
        }

        #region Implementation of IProvider

        public IStore GetStore(bool waitForInitialisation)
        {
            return _store;
        }

        public string HydraServer { get; private set; }

        public int? PollIntervalMs { get; private set; }

        public void ServerError(string server)
        {
            throw new NotImplementedException();
        }

        #endregion
    }
}

File: Java/hydra/src/test/java/uk/co/shastra/hydra/messaging/mocks/MockProvider.java
package uk.co.shastra.hydra.messaging.mocks;

import uk.co.shastra.hydra.messaging.storage.Provider;
import uk.co.shastra.hydra.messaging.storage.Store;

public class MockProvider implements Provider {

    private Store store;
	private String hydraServer;
	private Integer pollIntervalMs;

	public MockProvider(Store store) { this(store, null); }
	public MockProvider(Store store, Integer pollIntervalMs)
    {
        this.store = store;
        this.pollIntervalMs = pollIntervalMs;
        hydraServer = store.getName();
    }
    
	@Override
	public Store getStore(boolean waitForInitialisation) { return store; }

	@Override
	public String getHydraServer() { return hydraServer; }

	public Integer getPollIntervalMs() { return pollIntervalMs; }
	
	@Override
	public void serverError(String server) {
		// Not implemented
	}

}
